src/main/java/wooteco/subway/line/domain/Line.java

```diff
@@ -0,0 +1,119 @@
+package wooteco.subway.line.domain;
+
+import wooteco.subway.line.state.State;
+import wooteco.subway.line.state.StateFactory;
+import wooteco.subway.name.domain.LineName;
+import wooteco.subway.name.domain.Name;
+import wooteco.subway.name.domain.NullName;
+import wooteco.subway.station.domain.Station;
+
+import java.util.List;
+import java.util.Objects;
+
+public class Line {
+    private final Long id;
+    private Name name;
+    private String color;
+    private final State state;
+
+    public Line(final Long id) {
+        this(id, new NullName(), null);
+    }
+
+    public Line(final String name, final String color) {
+        this(null, new LineName(name), color);
+    }
+
+    public Line(final Long id, final String name, final String color) {
+        this(id, new LineName(name), color);
+    }
+
+    public Line(final Long id, final Name name, final String color) {
+        this(id, name, color, StateFactory.initialize(new Sections()));
+    }
+
+    public Line(final Long id, final String name, final String color, final List<Section> sections) {
+        this(id, new LineName(name), color, StateFactory.initialize(new Sections(sections)));
+    }
+
+    public Line(final Long id, final Name name, final String color, final State state) {
+        this.id = id;
+        this.name = name;
+        this.color = color;
+        this.state = state;
+    }
+
+    public Long id() {
+        return id;
+    }
+
+    public Name name() {
+        return name;
+    }
+
+    public String rawName() {
+        return name.name();
+    }
+
+    public String color() {
+        return color;
+    }
+
+    public Sections sections() {
+        return state.sections();
+    }
+
+    public List<Section> sortedSections() {
+        return state.sortedSections();
+    }
+
+    public void addSection(final Section targetSection) {
+        if (Objects.isNull(targetSection)) {
+            return;
+        }
+
+        if (state.sections().sections().size() < 1) {
+            this.state.sections().addSection(targetSection);
```

Comment:
sections를 꺼내서(getter) 추가하지 말고 객체에게 처리하라고 메세지를 보내세요

```suggestion
            this.state.addSection(targetSection);
```

Reply:
위 수정사항과 동일해요 😁

 9f38cc1